app.py

```diff
@@ -1,36 +1,62 @@
+import re
+import unicodedata
+import urllib
 import flask
 
 app = flask.Flask(__name__)
 
-ZWS = flask.Markup('<span class="highlight zero-width-char">zw</span>')
+character_names = { "\n": "\\n", "\r": "\\r", "\t": "\\t" }
 
-def replace_zero_width(text):
-  text = text.replace(u'\u200B', ZWS).replace(u'\u200C', ZWS)
-  return text.replace(u'\u200D', ZWS).replace(u'\uFEFF', ZWS)
+# Load some Unicode data remotely.
+unicode_blocks = []
+for line in urllib.urlopen("http://www.unicode.org/Public/UNIDATA/Blocks.txt").read().split("\n"):
```

Comment:
Good point. I still don't like to depend on an external service like this though. What if it's not available? The app crashes in a mysterious way.